javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1392,24 +1392,37 @@ public T getNext() {
             final Iterator<T> that = this;
             return new AbstractIterator<T>() {
 
-                private T next = null;
+                private T next;
```

Comment:
@paplorinc For now we follow the stateful design and low-level implementation of Scala's Iterator. See https://github.com/scala/scala/blob/2.12.x/src/library/scala/collection/Iterator.scala#L794-L823